polaris-react/src/components/ButtonGroup/ButtonGroup.scss

```diff
@@ -13,6 +13,7 @@
 }
 
 .Item {
+  display: flex;
```

Comment:
So, the reason I introduced this was because, we're wanting the buttons to be 24x24, but it's turning out to be 24x25. It seems this is due to the `line-height` being set to `normal` on the `Item` class of the `ButtonGroup`. It looks like it was introduced to combat this [problem](https://github.com/Shopify/polaris/issues/4079), where segmented buttons were not aligned properly.

I'll remove this `display` of `flex`, but propose we change the `line-height` from `normal` to `1`, so it fits the content inside of the button. However, I'm definitely open to other solutions.

I'm going to adjust the `ButtonGroup` storybook to also illustrate the case highlighted in the bug where text and icons are utilized within a `ButtonGroup`.

Reply:
Updated storybook
- [ButtonGroup](https://storybook.web.polaris-pagination-table-format.matt-kubej.us.spin.dev/?path=/story/all-components-buttongroup--with-segmented-buttons&globals=polarisSummerEditions2023:true;polarisSummerEditions2023ShadowBevelOptOut:true)
- [Pagination](https://storybook.web.polaris-pagination-table-format.matt-kubej.us.spin.dev/?path=/story/all-components-pagination--with-table-type&globals=polarisSummerEditions2023:true;polarisSummerEditions2023ShadowBevelOptOut:true)

Reply:
Normal is an odd decimal inconsistent across browsers. Line height of 1 is consistent and makes sense to me 👍🏽